src/cross_staff_offset.lua

```diff
@@ -2,26 +2,43 @@ function plugindef()
     finaleplugin.Author = "Carl Vine"
-    finaleplugin.AuthorURL = "http://carlvine.com"
-    finaleplugin.Version = "v1.21"
-    finaleplugin.Date = "2022/05/23"   
+    finaleplugin.AuthorURL = "http://carlvine.com/?cv=lua"
+    finaleplugin.Version = "v1.26"
+    finaleplugin.Date = "2022/07/09"
+    finaleplugin.AdditionalMenuOptions = [[  CrossStaff Offset No Dialog  ]]
+    finaleplugin.AdditionalUndoText = [[     CrossStaff Offset No Dialog  ]]
+    finaleplugin.AdditionalPrefixes = [[     cross_staff_nodialog = true  ]]
+    finaleplugin.AdditionalDescriptions = [[ Offset horizontal position of cross-staff note entries - NO DIALOG ]]
     finaleplugin.Notes = [[
-        When creating cross-staff notes using the option-downarrow shortcut, the stems of 'crossed' notes
-        are reversed (on the wrong side of the notehead) and so appear too far to the right (if shifting downwards)
-        by the width of one notehead (typically 24EVPU). This script lets you set a horizontal offset for all cross-staff
-        notes in the currently selected region, with a different offset for non-crossed notes. For crossing to the
-        staff below use (-24,0) or (-12,12). For crossing to the staff above use (24,0) or (12,-12). Also specify
-        which layer number (1-4) or "all layers" (0). (This also offers a simple way to reset the horizontal offset
-        of all notes in the selection to zero).
-
-        If you invoke the plugin with Option key (macOS) or Shift key pressed, the script uses the last settings
-        from the dialog.
-    ]]
-    return "CrossStaff Offset", "CrossStaff Offset", "Offset horizontal position of cross-staff note entries"
+        When creating cross-staff notes using the option-downarrow shortcut, the stems of 
+        'crossed' notes are reversed (on the wrong side of the notehead) and appear too far 
+        to the right (if shifting downwards) by the width of one notehead, typically 24EVPU. 
+        This script lets you set a horizontal offset for all cross-staff notes in the 
+        selected region, with a different offset for non-crossed notes.  
+        For crossing to the staff below use (-24,0) or (-12,12).  
+        For crossing to the staff above use (24,0) or (12,-12).  
+        Also specify which layer number to act upon (1-4) or "all layers" (0). 
+        (This also offers a simple way to reset the horizontal offset of all notes in the selection to zero).
+    
+        Under RGPLua (version 0.62+) the script adds an extra menu item that repeats the last 
+        chosen offset without presenting a dialog, for fast changes duplicating the last settings. 
+]]
+   return "CrossStaff Offset", "CrossStaff Offset", "Offset horizontal position of cross-staff note entries"
 end
- 
+
+local config = {
+    cross_staff_offset  = 0,
+    non_cross_offset    = 0,
+    layer_number        = 0,
+    script_name         = "cross_staff_offset"
+}
```

Comment:
Is it wise to include the script name in the configuration? It could potentially be changed by the configuration library, and I don't see the purpose of including it in the configuration. (FWIW: when the settings file is written, there is a comment at the top stating which script_name was passed in, and of course the filename itself contains the string as well.)

I just realized it would be possible to create a library function to return this by parsing it out of `finenv.RunningLuaFilePath()`. Is that something you would use instead?